theano/scan_module/scan_opt.py

```diff
@@ -1964,8 +1964,10 @@ def scan_merge_inouts(node):
             outputs = [outputs]
 
         na = scan_args(outer_inputs, outputs, op.inputs, op.outputs, op.info)
+        remove = [node]
     else:
         na = a
+        remove = []
```

Comment:
If this is not going to remove or replace anything, it should just return None.  Otherwise we are working for no reason.